HangFire.Autofac.Tests/AutofacJobActivatorTests.cs

```diff
@@ -178,6 +178,34 @@ public void InstanceRegisteredWith_InstancePerJob_IsDisposedOnScopeDisposal()
         }
 
         [TestMethod]
+        public void InstancePerJob_RegisteredWithExtraTags_ResolvesForJobScope()
+        {
+            var alternateLifetimeScopeTag = new object();
+            _builder.Register(c => new object()).As<object>()
+               .InstancePerBackgroundJob(alternateLifetimeScopeTag);
+            var activator = CreateActivator();
+
+            using (var scope = activator.BeginScope())
+            {
+                var instance = scope.Resolve(typeof(object));
+            }
+        }
+
+        [TestMethod]
+        public void InstancePerJob_RegisteredWithExtraTags_ResolvesForAlternateScope()
+        {
+            var alternateLifetimeScopeTag = new object();
+            _builder.Register(c => new object()).As<object>()
+               .InstancePerBackgroundJob(alternateLifetimeScopeTag);
+            var container = _builder.Build();
+
+            using (var scope = container.BeginLifetimeScope(alternateLifetimeScopeTag))
+            {
+                var instance = scope.Resolve(typeof(object));
```

Comment:
There's already tests that Hangfire.Autofac properly registers with its scope tag when needed and that the activator correctly instantiates new lifetime scopes. The only job of this change is to combine any incoming scope tags with hangfire's when registering, which is tested by being able to resolve from scopes of either lifetime and doesn't throw for a missing tag.

I could add that test in too if you want but you're really writing tests for Autofac itself at that point that its lifetime scopes work as it claims..